### PR TITLE
Fix numerical person properties turning to strings on save

### LINKED ISSUE #3189: Updating numerical properties via the UI changes type to string
## Bug description

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  

When updating the numerical properties of a user on the PostHog UI, they become strings once saved. 

## Expected behavior

If the prop was a number, we should try to convert it to a number upon change, if that fails, we can default to string.

## How to reproduce

1. Go to a person page
2. Update a numerical prop
3. Watch it become a string

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: _please provide_

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Close #3189

Tagging you @paolodamico because I wonder when we'll be releasing `persons-2353`. I'm not the biggest fan of all the UI updates but just the new logic is almost a necessity.

Spent more time than I'd like working on the old persons "logic" and decided it wasn't worth it from a prioritization standpoint.

Hence this fixes the logic on PersonV2 as I imagine we'll roll out the logic updates even if the experiment turns out bad from a UX perspective.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/scenes/persons/PersonV2.tsx [hunk 1]
```diff
@@ -147,7 +147,7 @@ function _PersonV2(): JSX.Element {
                     </Card>
                 </Col>
             </Row>
-            {mergeModalOpen && (
+            {mergeModalOpen && person && (
```
Comment (yakkomajuri) [inline frontend/src/scenes/persons/PersonV2.tsx:150]: this was a ts error

**Review by paolodamico — APPROVED** [review]
Great stuff, lgtm! Did a minor change so that we attempt to parse as number based on the new input not the old value. Please review before merging.

**Comment by paolodamico** [comment]
In a more general sense, I'll definitely check for the release of `persons-2353` this release cycle. Regardless I think it's definitely worth that you comment on the UI stuff that you don't like / would improve so we can work on those too if we do release it (#2353 or feel free to open a new issue).

**Comment by yakkomajuri** [comment]
makes sense, cheers!

**Pull request merged**